iis/iis.go

```diff
@@ -259,6 +265,42 @@ func applyAppPoolIdentity(appPoolName string, appPoolIdentity iisAppPoolIdentity
 	return nil
 }
 
+// Applies the Application Pool resource limit
+func applyAppPoolResourceLimit(appPoolName string, appPoolResource iisResourceLimit) error {
+	properties := []string{"set", "config", "-section:system.applicationHost/applicationPools"}
+
+	if appPoolResource.CPULimit != 0 {
+
+		cpuLimitSize := appPoolResource.CPULimit * 1000
+
+		properties = append(properties, fmt.Sprintf("/[name='%s'].cpu.limit:%s", appPoolName, strconv.Itoa(cpuLimitSize)))
+		properties = append(properties, fmt.Sprintf("/commit:apphost"))
+
+		if _, err := executeAppCmd(properties...); err != nil {
+			return fmt.Errorf("Failed to set Application Pool Resources: %v", err)
+		}
+	}
+	properties = []string{"set", "config", "-section:system.applicationHost/applicationPools"}
+
+	properties = append(properties, fmt.Sprintf("/[name='%s'].cpu.action:%s", appPoolName, "KillW3wp"))
+	properties = append(properties, fmt.Sprintf("/commit:apphost"))
+	if _, err := executeAppCmd(properties...); err != nil {
+		return fmt.Errorf("Failed to set Application Pool Resources: %v", err)
+	}
```

Comment:
I would prefer this be a part of the CPULimit scope above as this pertains to configuring the CPU. Does applying this settings cause a noop if the CPULimit is left at 0?